storage/zynq7000-flash/qspi.c

```diff
@@ -294,9 +295,19 @@ static void qspi_IOMode(void)
 
 	/* Configure controller */
 
-	/* Set baud rate to 100 MHz: 200 MHz / 2, master mode, not Legacy mode */
+	/* Set master mode, not Legacy mode */
+	*(qspi_common.base + cr) = 0x1 | (1 << 31);
+
 	*(qspi_common.base + cr) &= ~(0x7 << 3);
-	*(qspi_common.base + cr) |= 0x1 | (1 << 31);
+	if (QSPI_FCLK != -1) {
```

Comment:
Just my OCD, any negative value is invalid
```suggestion
	if (QSPI_FCLK >= 0) {
```